### PR TITLE
feat(checkbox): fix rendering in ion-item with new syntax

### PR DESCRIPTION
**Pull request by amandaejohnston**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the new behavior?

- `ionStyle` event now only emits with legacy checkbox to avoid styling issues when inside an item.
- Tests added for rendering a checkbox with the new syntax in an `ion-item`.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by liamdebeasi — APPROVED** [review]
Changes look good. Will the ripple effect be added in a separate PR?

**Comment by amandaejohnston** [comment]
Aha, didn't notice that had broken 👀 I've added it in here.

File: core/src/components/checkbox/checkbox.tsx [hunk 1]
```diff
@@ -245,6 +259,7 @@ export class Checkbox implements ComponentInterface {
             onBlur={() => this.onBlur()}
             ref={(focusEl) => (this.focusEl = focusEl)}
           />
+          {mode === 'md' && inItem && <ion-ripple-effect></ion-ripple-effect>}
```
Comment (liamdebeasi) [inline core/src/components/checkbox/checkbox.tsx:262]: IMO we should always render this in MD mode that way developers can use the ripple effect outside of `ion-item` if they want to

**Review by sean-perkins — APPROVED** [review]
This looks good to me. The failing test is flaky, which I've ran into as well:
>   1) [Mobile Safari] › src/utils/animation/test/multiple/animation.e2e.ts:16:3 › animation: multiple › should resolve grouped animations using css animations 

    Error: expect(received).toEqual(expected) // deep equality

    Expected: "AnimationCSubAFinished"
    Received: "AnimationCFinished"

We may want to just `test.skip` if re-running the action doesn't pass.

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
I'm noticing that the item's hover/focused/active effects are also applying: 
image (notice a lighter shade of gray on the left where the item is)

You likely need to update the `emitStyle` function to only run on a legacy checkbox

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
Also the checkbox does not take up the container's height in an item, so it looks misaligned: 
image


for context, this is what I did with toggle: https://github.com/ionic-team/ionic-framework/blob/16404ab12f4ef9095e57fbf31d7036ab7799b9f4/core/src/components/toggle/toggle.scss#L41-L44

edit: Discussed offline. I cannot reproduce this anymore.

File: core/src/components/checkbox/checkbox.tsx (hunk 1 above)
Comment (amandaejohnston) [reply core/src/components/checkbox/checkbox.tsx:262]: Checkboxes don't currently have a ripple effect with the legacy syntax, so I'm hesitant to change that. The [MD spec](https://m2.material.io/develop/web/components/input-controls/checkboxes) does show some different-looking hover/active/focus states, with the circle wrapping around the checkbox, but we don't have any of that in `main` either so making the change feels out of scope here.

**Review by liamdebeasi — CHANGES_REQUESTED** [review]
It looks like the hover/focus/activated effects are only applying on `ion-item` and not `ion-checkbox`.

It looks like you need to update the `hasCover` function in `ion-item` to only consider `ion-checkbox.legacy-checkbox`

**Comment by liamdebeasi** [comment]
Oh I think the same thing needs to be done for `setMultipleInputs` too (i.e. check for `ion-checkbox.legacy-checkbox`)

**Comment by amandaejohnston** [comment]
Removing the new CSS properties seems to have basically fixed all the item-related issues, so I was able to remove most of the new code here. Even the ripple effect is now properly handled by `ion-item` instead.

**Review by liamdebeasi — COMMENTED** [review]
Clicking the far left of the ion-item triggers the ripple effect but does not check the checkbox


https://user-images.githubusercontent.com/2721089/208174770-8a833604-6031-4ea1-a930-caf64767e9b4.mov


The rest of the changes look good though 👍

**Review by liamdebeasi — APPROVED** [review]
We discussed handling the ion-item margin issue separately as this behavior already exists in the latest version of Ionic 6.

**Review by sean-perkins — APPROVED** [review]

**Pull request merged**